File: Service/Flask-App/models/sector.py
#from users.specialist import Specialist

class Sector:
    def __init__(self, name, 
                       owner, 
                       website_url, 
                       address, 
                       longitude, 
                       latitude, 
                       b_h_open, 
                       b_h_close, 
                       founded, 
                       description,
                       isActive,
                       consultationFee,
                       contact,
                       email,
                       password):
        self.name = name
        self.owner = owner
        self.website_url = website_url
        self.address = address
        self.longitude = longitude
        self.latitude = latitude
        self.b_h_open = b_h_open
        self.b_h_close = b_h_close
        self.founded = founded
        self.description = description
        self.isActive = isActive
        self.consultationFee = consultationFee
        self.contact = contact
        self.email = email
        self.password = password
        self.specialists: list = []

    # def RegisterSpecialist(self, specialist: Specialist):
    #     self.specialists.append(specialist)

    # def getSpecialists(self):
    #     return self.specialists


    #setters
    def setName(self, name):
        self.name = name
    
    def setOwner(self, owner):
        self.owner = owner
    
    def setWebsite_url(self, url):
        self.website_url = url

    def setAddress(self, address):
        self.address = address

    def setLongitude(self, longitude):
        self.longitude = longitude
    
    def setLatitude(self, latitude):
        self.latitude = latitude
    
    def setBusinessHourseOpening(self, opening_time):
        self.b_h_open = opening_time

    def setBusinessHourseClose(self, closing_time):
        self.b_h_close = closing_time

    def setWhenFound(self, founded):
        self.founded = founded

    def setDescription(self, description):
        self.description = description


    def addSpecialist(self, specialist):
        self.specialists.append(specialist)


    #getters
    def getName(self):
        return self.name

    def getOwner(self):
        return self.owner

    def getWebsite(self):
        return self.website_url
    
    def getAddress(self):
        return self.address
    
    def getLongitude(self):
        return self.longitude

    def getLatitude(self):
        return self.latitude

    def getBusinessHourseOpening(self):
        return self.b_h_open
    
    def getBusinessHourseClose(self):
        return self.b_h_close

    def getWhenFound(self):
        return self.founded

    def getDesctription(self):
        return self.des

        

File: Client/eDoctor-Frontend/src/app/models/sector.ts
export class Sector {
  id = '';
  Name = '';
  Founder = '';
  Website = '';
  Address = '';
  Longitude = '';
  Latitude = '';
  B_Hours_Open = '';
  B_Hours_Close = '';
  Founded = '';
  Description = '';
  isActive = '';
  ConsultationFee: any = 0;
  Contact = '';
  Email = '';
  Approved = 0;
  Password = '';
  distance = 0;
  constructor(
    id: string,
    Name: string,
    Founder: string,
    Website: string,
    Address: string,
    Longitude: string,
    Latitude: string,
    B_Hours_Open: string,
    B_Hours_Close: string,
    Founded: string,
    Description: string,
    isActive: string,
    ConsultationFee: any = 0,
    Contact: string,
    Email: string,
    Approved: number,
    Password: string
  ) {
    this.id = id;
    this.Name = Name;
    this.Founder = Founder;
    this.Website = Website;
    this.Address = Address;
    this.Longitude = Longitude;
    this.Latitude = Latitude;
    this.B_Hours_Open = B_Hours_Open;
    this.B_Hours_Close = B_Hours_Close;
    this.Founded = Founded;
    this.Description = Description;
    this.isActive = isActive;
    this.ConsultationFee = ConsultationFee;
    this.Contact = Contact;
    this.Email = Email;
    this.Approved = Approved;
    this.Password = Password;
  }

  setDistance(distance: number) {
    this.distance = distance;
  }

  getDistance() {
    return this.distance;
  }
}
